EmbeddedScripts.Shared.Tests/ContainerTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Xunit;
+
+namespace EmbeddedScripts.Shared.Tests
+{
+    public class ContainerTests
+    {
+        [Fact]
+        public void RegisteringDependency_Succeed()
+        {
+            var x = 1;
+            var container = new Container();
+            container.Register(x, "x");
+            Assert.Equal(1, container.Resolve<int>("x"));
+        }
+
+        [Fact]
+        public void ResolveUnregisteredDependency_ThrowsException()
+        {
+            var container = new Container();
+            Assert.Throws<ArgumentException>(() => container.Resolve<int>("x"));
+        }
+
+        [Fact]
+        public void RevealAllAliases()
+        {
+            var container = new Container();
+
+            var x = 1;
+            var y = "2";
+
+            container.Register(x, "x");
+            container.Register(y, "y");
```

Comment:
Maybe more fluent way to setup dependencies/runners?
e.g. `container.Register(x, "x").Register(y, "y")`

Reply:
We can adjust it later. It can be a Builder pattern or registration Factory for the container.